langchain/chains/serverless/memory.py

```diff
@@ -0,0 +1,94 @@
+import boto3
+from langchain.chains.base import Memory
+from pydantic import BaseModel
+from typing import Any, Dict, List, Optional
+from datetime import datetime
+
+now = datetime.now()
+
+def _get_prompt_input_key(inputs: Dict[str, Any], memory_variables: List[str]) -> str:
+    # "stop" is a special key that can be passed as input but is not used to
+    # format the prompt.
+    prompt_input_keys = list(set(inputs).difference(memory_variables + ["stop"]))
+    if len(prompt_input_keys) != 1:
+        raise ValueError(f"One input key expected got {prompt_input_keys}")
+    return prompt_input_keys[0]
+
+class ConversationBufferDynamoDBMemory(Memory, BaseModel):
+    """Buffer for storing conversation memory in DynamoDB."""
+
+    human_prefix: str = "Human"
+    ai_prefix: str = "AI"
+    """Prefix to use for AI generated responses."""
+    output_key: Optional[str] = None
+    input_key: Optional[str] = None
+    memory_key: str = "chat_history"
+    table_name: str = "ConversationHistory"
+    buffer: str = ""
```

Comment:
See my comment about making this a DynamoDB buffer.

```python
buffer: DynamoDBBuffer
```

Reply:
After attempting to implement this I've run into a question. Currently the implementation is invoked as follows:
`_id = 'testing_new_class'
table_name = 'ConversationTable'
memory = ConversationBufferDynamoDBMemory(id=_id, table_name=table_name)`

Yet under the class you suggest I assume it would have to be invoked like this instead:

`session_id = 'testing_new_class'
table_name = 'ConversationTable'
buffer = DynamoDBBuffer(table_name)
memory = ServerlessMemory(buffer=buffer, session_id=session_id)`

Just confirming that this is the implementation you have in mind.

Reply:
@KBB99 
Thanks for looking into this.

Langchain is moving so fast 🚀. The memory classes have been updated since my last comment, so you might have to update based on these new constructs. See here: https://langchain.readthedocs.io/en/latest/modules/memory/getting_started.html 

I am already looking at the new updates in Langchain, and can post an update to your PR early next week. Let me know if you are ok with me doing this.

Reply:
Oh boy, you're right. I didn't realize the underlying memory classes had changed. I will attempt to change my ServerlessMemory class to work with the new ones.

Reply:
Luckily making the adjustment was easy and just required updating the using the new BaseMemory class from Langchain.schemas. I am good with you doing whatever you see fit :).